_posts/2016-03-07-optimized-php.markdown

```diff
@@ -0,0 +1,56 @@
+
+Indeed, more the binary to be used in a subprocess is small, more the creation of the corresponding sub-process occurs quickly. However, by default, the binary PHP command line contain on most cases a number of modules that are not necessarily useful to run the tests. To convince you, simply retrieve the list of modules into your PHP executable with the command *php -m*. Then you will definitely notice that most of them are completely unnecessary to the proper execution of your tests. And moreover, that it's possible to turn them off when PHP is compiled in order to obtain a more compact binary. But there is a price to pay for it, since you then must compile *PHP* manually. it's however not too expensive, because the procedure for this is relatively simple.
+
+Traditionally, once sources of language retrieved via [php.net](http://www.php.net/) compiling PHP is done as follows under UNIX:
+{% highlight shell %}
+	# cd path/to/php/source
+	# ./configure
+	# make
+	# make install
+{% endhighlight %}
+
+Note that the command *make install* must be executed as root to function properly. Because we want a custom version of PHP, it's necessary to change this procedure at the level of the command *./configure*. It's this command that allows to define, among other things, the modules that will be integrated to PHP in its compilation, like we see in the output of the command *./configure --help*.
+
+To get a version of PHP corresponding exactly to your needs, you need to start by disable all of the default modules, via the option *--disable-all*. Once this done, it's required to add the option *--enable-cli* to get only, at the end of compilation, a usable PHP binary. Then add via the options * --enable-* * the modules required for the execution of your tests, as well as any options * --with-* * necessary to compile these modules. For example, the command to use to compile a binary PHP command-line necessary and sufficient to run the unit tests of atoum in Mac OS X is:
+
+{% highlight shell %}
+	# ./configure --disable-all --sysconfdir=/private/etc --enable-cli --with-config-file-path=/etc --with-libxml-dir=/usr  --with-pcre-regex --enable-phar --enable-hash --enable-json --enable-libxml --enable-session --enable-tokenizer --enable-posix --enable-dom
+{% endhighlight %}
+
+Noted that if you wish to install your PHP binary to a specific location to avoid replacing the one already installed at the level of your operating system, you can use the option *-prefix = path/to/destination/directory*, as shown in the help of *./configure*, available via the option *-help*. You can then use it to run your tests via the argument *-p* of atoum.
```

Comment:
`s/*-prefix = path/to/destination/directory*/`--prefix = path/to/destination/directory`/`